### ISSUE TITLE
[Announcement] Moving repos to mono-repos

### ISSUE THREAD

**Issue by mosuem** [dart-lang/sdk#56591]
To ease the maintenance of the dart-lang/ repositories, we are currently moving some of them into monorepos.

Our thematically grouped monorepos are:

- [build](https://github.com/dart-lang/build/)
- [test](https://github.com/dart-lang/test/)
- [native](https://github.com/dart-lang/native/)
- [http](https://github.com/dart-lang/http/)
- [webdev](https://github.com/dart-lang/webdev/)
- [i18n](https://github.com/dart-lang/i18n/)

And the publisher grouped, more miscellaneous are:

- [tools](https://github.com/dart-lang/tools/)
- [core](https://github.com/dart-lang/core/)
- [labs](https://github.com/dart-lang/labs/)

Progress tracker:

- [x] coverage -> tools
- [x] oauth2 -> tools
- [x] source_map_stack_trace -> tools
- [x] mime -> tools
- [ ] crypto -> core

Please report any issues you are facing with this here.

**Comment by lrhn** [comment]
If possible, please ensure that existing issues (at least open issues) and pull requests are migrated along with the code. 

That's likely non-trivial, also because of references from other issues, potentially in other repositories. Rewriting all of those ... should be possible. If GitHub can show "Referenced from" in an issue, we should be able to find that issue and rewrite it after copying the original issue to a new location.
PRs likely need something tricky too. Or maybe just a final merge commit. (May still risk including the entire old repository in the new repository, as the parent of a PR commit.)

**Comment by mosuem** [comment]
> If possible, please ensure that existing issues (at least open issues) and pull requests are migrated along with the code.

Yes, that is being done. This is a Github feature, it should work well. The PRs referenced in commit messages are also rewritten to reference the correct PRs from the old repo.

**Comment by devoncarew** [comment]
Just for posterity, our reasons for consolidating packages into fewer repos are:

- economies of scale; we find the maintence costs mostly scale with the number of repositories
- ability to invest more in tooling; we're able to more easily roll out things like publishing automation, health checks, and other tools with fewer repos. We can concentrate more on those fewer repos.
- better maintenance; we have fewer PR queues to monitor and generally less to track

And as an update on our progress, all the packages we plan to move into `dart-lang/core` have been moved. About half of those planned for `dart-lang/tools` have been moved, and we have outstanding PRs for the balance. We'll close this issue once the bulk of the moves have been performed; any consolidation after that will be opportunistic.

**Comment by devoncarew** [comment]
This work is largely complete.
